mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -212,12 +226,110 @@ class LinearRegressionModel private[ml] (
   extends RegressionModel[Vector, LinearRegressionModel]
   with LinearRegressionParams {
 
+  @transient private var trainingResults: Option[LinearRegressionTrainingResults] = None
+
+  /**
+   * Gets results (e.g. residuals, mse, r^2) of model on training set. This method should only
+   * be called on the driver (it is not available on workers).
+   */
+  def getTrainingResults: Option[LinearRegressionTrainingResults] = trainingResults
```

Comment:
Option isn't very Java-friendly, so providing ```summary``` and ```hasSummary``` might be best.  You could keep ```getSummary``` as the Scala version which returns an Option.